SpockOpaleye/src/TenantApi.hs

```diff
@@ -58,50 +40,25 @@ deactivate_tenant :: Connection -> Tenant -> IO Tenant
 deactivate_tenant conn tenant = set_tenant_status conn tenant TenantStatusInActive
 
 set_tenant_status :: Connection -> Tenant -> TenantStatus -> IO Tenant
-set_tenant_status conn tenant status = update_tenant conn (tenant_id tenant)
-                                              tenant { tenant_status = status }
+set_tenant_status conn tenant st = update_tenant conn (tenant ^. id) (tenant & status .~ st)
 
 update_tenant :: Connection -> TenantId -> Tenant -> IO Tenant
```

Comment:
please see my earlier comment about passing `id` separately from the record (which already contains the ID).